r-package/grf/vignettes/rate.Rmd

```diff
@@ -0,0 +1,216 @@
+
+```{r}
+# Read in dummy data from https://github.com/grf-labs/grf/tree/master/r-package/grf/vignettes
+load("synthetic_SPRINT_ACCORD.RData")
+df <- data.frame(Y = c(Y.sprint, Y.accord),
+                 D = c(D.sprint, D.accord),
+                 data = c(rep("synthetic-SPRINT", length(Y.sprint)),
+                          rep("synthetic-ACCORD", length(Y.accord))))
+df$Censored <- factor(df$D, labels = c("Yes", "No"))
+
+ggplot(df, aes(x = Y, fill = Censored)) +
+  facet_wrap(data ~ .) +
+  geom_histogram(alpha = 0.5, bins = 30) +
+  xlab("Time until primary outcome (days)") +
+  ylab("Frequency") +
+  theme_classic()
+```
+
+The SPRINT trial was halted early due to a low number of events occurring and thus have a very high censoring rate. The target estimand we consider in this example is the difference in restricted mean survival time (RMST) conditional on covariates:
+
+$$\tau(x) = E[T(1) \land h - T(0) \land h \, | X = x],$$
+where $T$ is the (censored) survival time and we set $h$ (`horizon` in the GRF package) to ~ 3 years, after which the SPRINT trial data nearly stops exhibiting events. Since the treatment was randomized we set the propensity scores `W.hat`$=E[W_i \, | X_i = x]$ to the mean number of treated units.
+
+```{r}
+horizon <- 3 * 365
+csf.sprint <- causal_survival_forest(X.sprint, Y.sprint, W.sprint, D.sprint,
+                                     W.hat = mean(W.sprint), target = "RMST", horizon = horizon)
+
+csf.accord <- causal_survival_forest(X.accord, Y.accord, W.accord, D.accord,
+                                     W.hat = mean(W.accord), target = "RMST", horizon = horizon)
+
+tau.hat.accord <- predict(csf.sprint, X.accord)$predictions
+tau.hat.sprint <- predict(csf.accord, X.sprint)$predictions
+```
+
+Evaluating RATE on SPRINT and ACCORD, using estimated CATE functions from ACCORD and SPRINT then gives
+
+```{r}
+rate.sprint <- rank_average_treatment_effect(csf.sprint, tau.hat.sprint, target = "AUTOC")
+rate.accord <- rank_average_treatment_effect(csf.accord, tau.hat.accord, target = "AUTOC")
+
+rate.sprint
+rate.accord
+```
+
+```{r}
+par(mfrow = c(1, 2))
+plot(rate.sprint, xlab = "Treated fraction", main = "TOC evaluated on SPRINT\n tau(X) estimated on ACCORD")
+plot(rate.accord, xlab = "Treated fraction", main = "TOC evaluated on ACCORD\n tau(X) estimated on SPRINT")
+```
+
+In this toy example there is no evidence of significant HTEs in the two trials (note: this can also be attributed to a) low power, as perhaps the sample size is not large enough to detect HTEs, or b) that the HTE estimator does not detect them). For a broader analysis comparing different prioritization strategies and CATE estimators see Yadlowsky et al., (2021).
```

Comment:
Perhaps also mention that it could be because the heterogeneity in the treatment effects along observable predictor variables are negligible.

Reply:
Done. Maybe also be extra careful and make sure this final sentence is 100% agreeable as I can imagine this is something many applied people will take as the literal conclusion after running RATE and copy/paste as appropriate

Also, the RATE paper didn't compute QINI for this example, I can imagine some people will ask why not?